git-cliff/src/lib.rs

```diff
@@ -273,10 +281,34 @@ fn process_repository<'a>(
 	let mut previous_release = Release::default();
 	let mut first_processed_tag = None;
 	let repository_path = repository.path()?.to_string_lossy().into_owned();
+	let mut submodule_commits: HashMap<PathBuf, (&mut Repository, Vec<Commit>)> =
```

Comment:
Why do we have this complex type? 🤔

Reply:
So we don't have to reinit The repo every time we want to parse commits, but tbh I put some of that together after a very full day so I plan on redoing it a bit. The pathbuf being the key is just to avoid having to impl comparisons for submodules

Reply:
Got it. it would be nice to refactor/simplify this a bit for sure :)

Reply:
I think I might just swap it to be a tuple and replace the end commit instead. Plan is that for any given submodule, the releases we include should be inclusive and in aggregate. So if a single release has multiple commits bumping the submodule underneath it, just aggregate all of those changes